### ISSUE TITLE
Reasons to blacklisting

### ISSUE THREAD

**Issue by Sinnoma** [Cog-Creators/Red-DiscordBot#3128]
# Feature request

#### Type of feature you are requesting:

- [ ] Cog
- [X] Command
- [ ] API functionality

#### Describe your requested feature

When we are blacklisting a user on the bot blacklist, we could be able to add an reason (Or not, if ommited, the reason will be `No reason(s) provided.`) that will be showed on the blacklisted users list. 

For actual blacklists (I mean, if it's added and when the user will update his instance), there will be no reasons.

Also add a command to add a reason after blacklisting, in case if the owner forgot to add one.

**Comment by Drapersniper** [comment]
Sounds like you should ban the people instead of blacklist them if they are causing you enough trouble to want to blacklist them and remember the reason

**Comment by Vexed01** [comment]
The main issue is that blacklist is an owner only command, so it would have to be a global reason; and, as an owner, you could keep track of it yourself. _If_ blacklist was server-based and permitted server owners/admins/mods to blacklist users, then this may make sense to integrate into the core modlog cog.
The reason I am saying to keep track yourself is that there isn't a need to coordinate with other moderators/admins and hold them accountable as only one person, the bot owner, can blacklist people.
TLDR: Only you can blacklist so you should be able to keep track yourself.

> #feature-ideas
[I mean, if he want to ask an unblacklist (And he if receving the DM), he could ask to the owner/co-owner](https://discordapp.com/channels/133049272517001216/133049878837329920/645233173491613696)

As you implied that you have coowner(s), I do understand that you may want to keep track of who blacklisted who and why without relying on coowner(s) telling you personally. However, as coowner isn't a fully supported feature this is a niche case and unlikely to be added to the core bot. You may, however, ask for a [cog bounty](https://cogboard.red/c/bounties) (which may require an API functionality request to the core bot, depends on how it's written).

Edit:
`[p]localblacklist` exists. Sorry...
I do still think that this is a niche case, though, and is unlikely to be implemented.

**Comment by Sinnoma** [comment]
Better to close it and let that down...
